### DISCUSSION TITLE
Query timestamps lead to empty patch

### DISCUSSION THREAD

**Discussion by Modexus** [torchgeo/torchgeo#645] [General]
Given the warping of tiles (#548) and the handling of the `query` `mint` and `maxt` values even tile based datasets (NAIP) return empty patches sometimes.

`(mint, maxt)` are based on the `hit` but different tiles have different capture dates which lead to queries that the original `hit` does not contain (because of warping) but the next tile is not returned in the intersect as it has a non-intersecting `(mint, maxt)` with original hit.

It may make sense to set `(mint, maxt)` always to the values in the `roi` instead of intersecting with the `hit`.

**Comment by adamjstewart** [comment]
> It may make sense to set `(mint, maxt)` always to the values in the `roi` instead of intersecting with the `hit`.

I think this depends strongly on the particular application you are working on. For example, if you are working with time series data (https://github.com/microsoft/torchgeo/discussions/640), then this solution wouldn't be useful. In general I think we need more builtin samplers to handle these kinds of situations, one of which could very well include the suggestion you describe. I'm going to move this to a discussion since it's not exactly a bug (it's by design).
